dev/com.ibm.ws.sipcontainer/src/com/ibm/ws/sip/stack/transport/sip/netty/SipConnLink.java

```diff
@@ -0,0 +1,523 @@
+
+	/**
+	 * queue of messages waiting to be sent out to the network. the first
+	 * element in this list is currently being sent.
+	 */
+	final LinkedList<MessageContext> m_outMessages;
+
+	/** true if the last sent message has not completed yet */
+	private boolean m_sendPending;
+
+	/** stateful message parser dedicated to this stream connection */
+	private MessageParser m_messageParser;
+
+	/** buffer size for reading inbound messages */
+	private static final int READ_BUFFER_SIZE = 2048;
+
+	/**
+	 * exception provided by the error() callback, indicating the connection is
+	 * broken, and at least one message is waiting for writeComplete() or
+	 * writeError(). if writeError() is called, it takes care of cleanup. if
+	 * writeComplete() is called, it calls connectionError() with this
+	 * exception.
+	 */
+	private IOException m_readError;
+
+	/** true if waiting for pending outbound messages to go before closing down */
+	private boolean m_closing;
+
+	/** true if an error occurred on this connection, that requires closing it down */
+	private boolean m_broken;
+	
+	/** Netty channel associated with that SIP connection */
+	protected Channel m_channel;
+
+	/** the maximum number of messages waiting in the outbound queue */
+	private static final int s_maxOutboundPendingMessages =
+		SIPTransactionStack.instance().getConfiguration().getMaxOutboundPendingMessages();
+
+	/**
+	 * constructor for inbound connections, or for udp
+	 * 
+	 * @param sipInboundChannel
+	 *            channel that created this connection
+	 * @param channel
+	 *            Netty channel associated with that SIP connection
+	 */
+	public SipConnLink(SipInboundChannel sipInboundChannel, Channel channel) {
+		this(null, 0, sipInboundChannel, channel);
+	}
+
+	/**
+	 * constructor for outbound connections
+	 * 
+	 * @param peerHost
+	 *            remote host address in dotted form
+	 * @param peerPort
+	 *            remote port number
+	 * @param channel
+	 *            channel that created this connection
+	 */
+	public SipConnLink(String peerHost, int peerPort, SipInboundChannel sipInboundChannel, Channel channel) {
+		super(peerHost, peerPort, sipInboundChannel);
+		m_channel = channel;
+		m_outMessages = new LinkedList<MessageContext>();
+		m_sendPending = false;
+		m_messageParser = new StreamMessageParser(this);
+		m_readError = null;
+		m_closing = false;
+		m_broken = false;
+	}
+
+	/**
+	 * called by the derived class when connection established, either inbound
+	 * or outbound
+	 */
+	protected void connectionEstablished() {
+		
+	  if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+            Tr.debug(tc, "connectionEstablished");
+        }
+		  
+		
+		logConnection();
+		// 1. in case outbound messages queued while trying to connect, send
+		// them now
+		synchronized (m_outMessages) {
+			super.connectionEstablished();
+			sendPendingMessages();
+		}
+
+		// 2. prepare for reading inbound messages
+		/*
```

Comment:
Is this a missing TODO?

Reply:
Think this is a TODO brought over from the original chfw impl

Reply:
yea, carry over, but I've deleted this whole commented block